fcrepo-http-api/src/main/java/org/fcrepo/http/api/responses/StreamingBaseHtmlProvider.java

```diff
@@ -236,32 +210,17 @@ private Template getTemplate(final Model rdf, final Node subject,
                                   .findFirst();
 
         if (!template.isPresent()) {
-            LOGGER.trace("Attempting to discover mixin types of node for resource in question: {}", subject);
-            template = rdf.listObjectsOfProperty(createResource(subject.getURI()),
-                                                 createProperty(getRDFNamespaceForJcrNamespace(JCR_NAMESPACE) +
-                                                     "mixinTypes"))
-                          .toList().stream()
-                          .map(x -> x.asLiteral().getLexicalForm())
+            LOGGER.trace("Attempting to discover RDF types of node for resource in question: {}", subject);
+            template = ofNullable(getFirst(iter(rdf.listObjectsOfProperty(subjectURI, type))
```

Comment:
I know, I'm saying I would resolve that monad immediately. But it's not a big thing. If you think that would be less clear, fine.